components/studio/charts/apps/rstudio/chart/templates/deployment.yaml

```diff
@@ -0,0 +1,74 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ .Values.appname }}
+  namespace: {{ .Values.namespace }}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      release: {{ $.Release.Name }}
+      app: rstudio
+      project: {{ $.Values.project.slug }}
+      type: app
+      pod: rstudio
+  template:
+    metadata:
+      annotations:
+        prometheus.io/scrape: "true"
+        prometheus.io/path: /metrics
+        # prometheus.io/port: "8501"
+      labels:
+        release: {{ $.Release.Name }}
+        app: rstudio
+        project: {{ $.Values.project.slug }}
+        type: app
+        pod: rstudio
+    spec:
+    {{ if .Values.app.gpu.enabled }}
+      affinity:
+        {{ .Values.app.affinity | toYaml | nindent 8 | trim }}
+    {{ end }}
+    {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+    {{- end }}
+      containers:
```

Comment:
Where is the `securityContext`?

You need to make sure that the pod has no elevated permissions, do not run as root etc. The best is also to make the whole container read-only, adding `emptyDir`s where you need to be able to write (you should anyway have `emptyDir`s for those folders for performance reasons, even if they are not read-only). The default `securityContext` I use for pod-level is:
```
          securityContext:
            allowPrivilegeEscalation: false
            privileged: false
            runAsNonRoot: true
            runAsUser: 10500
            readOnlyRootFilesystem: true
            capabilities:
              drop:
                - all
            runAsGroup: 10500
```